scripts/lua/lib/dataStore.lua

```diff
@@ -0,0 +1,107 @@
+
+
+
+_M = {} 
+-- this script dynamically loads a persistence plugin and uses it to make calls to the persistent datastore
+
+
+local DATASTORE = os.getenv( 'DATASTORE')
+local utils = require('lib/utils')
+
+if DATASTORE == nil then 
+  DATASTORE = 'redis'
+end
+
+local impl = require(utils.concatStrings({'lib/', DATASTORE}))
+
+function _M.init() 
+  return impl.init()
+end 
+
+function _M.close(ds)
```

Comment:
hmm I think that should be fine, the only thing I would worry about is storing it in the module gets scoped to the worker process rather than the request, but I think if we are doing init every time anyways that should be fine. I'll try it out